### DISCUSSION TITLE
Scripting questions about StelUtils functions

### DISCUSSION THREAD

**Discussion by GFrmnt** [Stellarium/stellarium#3558] [Q&A]
Hello ! I am developing some scripts for a science museum and I have some question.
I would like to use the functions provided by the StelUtils.hpp file inside my script (to convert some data, for example).
Is there a way to do that ? Should I write something before the function like "core." ?

Have a nice day !

**Comment by gzotti** [comment]
StelUtils is not scriptable. The available commands of the core object are described in the scripting API documentation on stellarium.org. What is not described there is not available. Which functions from StelUtils would you need us to add to the core object?

**Reply by GFrmnt** [reply]
Thanks a lot for the quick reply !

I'm currently working on a script that would allow a video to appear transparently and create personalized animations between the stars in the Stellarium sky.

To do this, I may need to easily convert equatorial coordinates into the X Y position of my screen (with the "eclToEqu" function of the StelUtils and the "getScreenXYFromAltAzi" that I can already use for script)

There may already be other ways of doing this, but I'm still a beginner.

**Reply by gzotti** [reply]
You could search for some object close to your equatorial coordinates and retrieve its data which contain az/alt coordinates.

**Reply by GFrmnt** [reply]
Ok thanks a lot ! I will try this, but I'm trying to adapt the video display to the present date and I don't think it wil be possible like that.

 And there is no option to use function provided by the math.h librairy in python like cos() inside the script engine ?

**Reply by gzotti** [reply]
I am no JavaScript expert, but I think something like "include Math" exists. Then just re-implement the short conversion functions yourself in JS. (IIRC, `Math.cos(x)` etc.)
